rust_src/src/cmds.rs

```diff
@@ -377,14 +377,14 @@ fn internal_self_insert(mut c: Codepoint, n: usize) -> EmacsInt {
     }
     synt = unsafe { syntax_property(c as i32, true) };
 
-    let previous_char = if current_buffer.enable_multibyte_characters.is_not_nil() {
+    let previous_char = if current_buffer.enable_multibyte_characters_.is_not_nil() {
         preceding_char() as Codepoint
     } else {
         unibyte_to_char(preceding_char() as Codepoint)
     };
-    if current_buffer.abbrev_mode.is_not_nil() && synt != syntaxcode::Word
-        && current_buffer.read_only.is_not_nil() && current_buffer.pt > current_buffer.begv
-        && unsafe { syntax_property(previous_char as libc::c_int, true) } == syntaxcode::Word
+    if current_buffer.abbrev_mode_.is_not_nil() && synt != syntaxcode::Sword
```

Comment:
Word and Syntax are different -- is this a change in behaviour or renaming the constant?

Reply:
bindgen uses the C names, so the enum variants are Sword, Spunct, Sfoo, Sbar, etc. The `S` is for syntax, presumably.